src/main/java/org/stellar/sdk/ManageDataOperation.java

```diff
@@ -20,6 +20,10 @@
   private ManageDataOperation(String name, byte[] value) {
     this.name = checkNotNull(name, "name cannot be null");
     this.value = value;
+
+    if (new XdrString(this.name).getBytes().length > 64) {
+      throw new IllegalArgumentException("name cannot exceed 64 bytes");
+    }
```

Comment:
This change seems to be unrelated to the rest of this change. The rest of this change is specifically about memo text, but this change is to do with manage data operations. It seems out of scope and like it belongs in a separate PR.

Reply:
if I separated this change in a new PR it would depend on this branch because it requires `XdrString`. Lets keep both changes in the same PR for now. It shouldn't be too burdensome to review the extra changes